kvdb/src/lib.rs

```diff
@@ -129,3 +149,20 @@ pub trait KeyValueDB: Sync + Send + parity_util_mem::MallocSizeOf {
 		IoStats::empty()
 	}
 }
+
+/// Return for a start inclusive prefix, the non inclusive end.
+pub fn end_prefix(prefix: &[u8]) -> Vec<u8> {
```

Comment:
this function can be simplified
```rust
fn end_prefix(mut prefix: Vec<u8>) -> Vec<u8> {
    while let Some(0xff) = prefix.last() {
        prefix.pop();
    }
    if let Some(byte) = prefix.last_mut() {
        *byte += 1;
    }
    prefix
}
```
I'm not sure we want to expose it `kvdb`, I see it's used as a utility function in `kvdb-rocksdb` and `kvdb-memorydb`, but for example, we don't expose `other_io_error`

Reply:
Also if the prefix is empty or 0xfffffff...ff, will it delete the whole db?

Reply:
And we could reuse this function for iter_from_prefix by setting https://docs.rs/rocksdb/0.13.0/rocksdb/struct.ReadOptions.html#method.set_iterate_upper_bound

Reply:
for empty prefix it does indeed delete every values (not in a very efficient way).

Reply:
What about `0xff` prefix? I think we should add some tests and document the difference between `0xff` and `0xff00` as it's not intuitive.

Reply:
I am not sure how to document that, I will add a test it may make thing more explicit.

Reply:
oups, I actually meant `0xff` and `0x00ff`

Reply:
> I am not sure how to document that,

Let's add a few examples and include interesting cases such as `0xff` and `0xff00` among the examples?